pkg/api/api_test.go

```diff
@@ -180,11 +256,191 @@ var _ = Describe("Api", func() {
 				Expect(err).NotTo(HaveOccurred())
 
 				rr := httptest.NewRecorder()
-				handler := http.HandlerFunc(catalogAPI.ProfileHandler)
+				handler := http.HandlerFunc(catalogAPI.ProfileWithVersionHandler)
 
 				handler.ServeHTTP(rr, req)
 
 				Expect(rr.Code).To(Equal(http.StatusNotFound))
+				Expect(fakeCatalog.GetWithVersionCallCount()).To(Equal(1))
+				actualSourceName, actualProfileName, actualCatalogVersion := fakeCatalog.GetWithVersionArgsForCall(0)
+				Expect(actualSourceName).To(Equal(sourceName))
+				Expect(actualProfileName).To(Equal(profileName))
+				Expect(actualCatalogVersion).To(Equal(version))
+			})
+		})
+
+		When("a querry param is missing", func() {
+			BeforeEach(func() {
+				fakeCatalog.GetWithVersionReturns(nil)
+			})
+
+			Context("catalog param", func() {
+				It("returns a 404", func() {
+					req, err := http.NewRequest("GET", "/profile/catalog/nginx-1/v0.3.0", nil)
+					req = mux.SetURLVars(req, map[string]string{"profile": profileName, "version": version})
+					Expect(err).NotTo(HaveOccurred())
+
+					rr := httptest.NewRecorder()
+					handler := http.HandlerFunc(catalogAPI.ProfileWithVersionHandler)
+
+					handler.ServeHTTP(rr, req)
+
+					Expect(rr.Code).To(Equal(http.StatusBadRequest))
+					Expect(fakeCatalog.GetWithVersionCallCount()).To(Equal(0))
+				})
+			})
+
+			Context("profile param", func() {
+				It("returns a 404", func() {
+					req, err := http.NewRequest("GET", "/profile/catalog/nginx-1/v0.3.0", nil)
+					req = mux.SetURLVars(req, map[string]string{"catalog": sourceName, "version": version})
+					Expect(err).NotTo(HaveOccurred())
+
+					rr := httptest.NewRecorder()
+					handler := http.HandlerFunc(catalogAPI.ProfileWithVersionHandler)
+
+					handler.ServeHTTP(rr, req)
+
+					Expect(rr.Code).To(Equal(http.StatusBadRequest))
+					Expect(fakeCatalog.GetWithVersionCallCount()).To(Equal(0))
+				})
+			})
+
+			Context("version param", func() {
+				It("returns a 404", func() {
+					req, err := http.NewRequest("GET", "/profile/catalog/nginx-1/v0.3.0", nil)
+					req = mux.SetURLVars(req, map[string]string{"profile": profileName, "catalog": sourceName})
+					Expect(err).NotTo(HaveOccurred())
+
+					rr := httptest.NewRecorder()
+					handler := http.HandlerFunc(catalogAPI.ProfileWithVersionHandler)
+
+					handler.ServeHTTP(rr, req)
+
+					Expect(rr.Code).To(Equal(http.StatusBadRequest))
+					Expect(fakeCatalog.GetWithVersionCallCount()).To(Equal(0))
+				})
+			})
+		})
+	})
+
+	Context("/profiles/catalog/profile-name/version/available_updates", func() {
```

Comment:
this entire context was missing :face_with_head_bandage:

Reply:
Oh crap!! We didn't add a test on the api!! AAAA.